LEGO1/omni/src/video/mxbitmap.cpp

```diff
@@ -207,14 +208,83 @@ MxResult MxBitmap::LoadFile(HANDLE p_handle)
 	return result;
 }
 
-// STUB: LEGO1 0x100bce70
-void MxBitmap::VTable0x2c(int, int, int, int, int, int, int)
+// FUNCTION: LEGO1 0x100bce70
+void MxBitmap::BitBlt(
+	MxBitmap* p_src,
+	MxS32 p_srcLeft,
+	MxS32 p_srcTop,
+	MxS32 p_dstLeft,
+	MxS32 p_dstTop,
+	MxS32 p_width,
+	MxS32 p_height
+)
 {
+	MxLong dstHeight = AbsFlipped(m_bmiHeader->biHeight);
+	MxLong srcHeight = AbsFlipped(p_src->m_bmiHeader->biHeight);
+	if (GetRectIntersection(
+			p_src->m_bmiHeader->biWidth,
+			srcHeight,
+			m_bmiHeader->biWidth,
+			dstHeight,
+			&p_srcLeft,
+			&p_srcTop,
+			&p_dstLeft,
+			&p_dstTop,
+			&p_width,
+			&p_height
+		)) {
+		MxU8* srcStart = p_src->GetStart(p_srcLeft, p_srcTop);
+		MxU8* dstStart = GetStart(p_dstLeft, p_dstTop);
+		MxLong srcStride = p_src->GetAdjustedStride();
+		MxLong dstStride = GetAdjustedStride();
+		for (; p_height; --p_height) {
```

Comment:
This feels a little weird, I'll investigate later

Reply:
Maybe it's a while loop instead. There isn't much difference in compiled code.